lib/use_cases/view_historical_percent_changes.rb

```diff
@@ -0,0 +1,32 @@
+class ViewHistoricalPercentChanges
+  def initialize(rent_charge_gateway:, fixed_data_gateway:)
+    @rent_charge_gateway = rent_charge_gateway
+    @fixed_data_gateway = fixed_data_gateway
+  end
+
+  attr_reader :rent_charge_gateway,
+              :fixed_data_gateway
+
+  def execute
+    counts.map do |item|
+       { year: item[0], no_of_accounts: item[1], rc_uplift: fixed_data_gateway.rc_uplift(item[0]) }
+    end
+  end 
+
+  def percent_changes_for_specific_year(year)
+    return { error: :no_record_found } if no_of_accounts_count(year).zero?
+    { year: year, no_of_accounts: no_of_accounts_count(year), rc_uplift: fixed_data_gateway.rc_uplift(year) }
+  end 
+
+  private
+
+  def counts
+    rent_charge_gateway.all.group_by { |i| i[:year] }.map do |key, value| 
+      [key , value.count]
```

Comment:
I wonder if returning a hash here would be more expressive? Then in `execute` you'd be able to write `item[:year]` and `item[:count]` rather than `item[0]` and `item[1]`. Would also make this easier to change if we wanted to add more data.